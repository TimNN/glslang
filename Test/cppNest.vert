#define ON

float sum = 0.0;

void main()
{

#ifdef ON
//yes
sum += 1.0;

    #ifdef OFF
    //no
    sum += 20.0;
    #endif

    #if defined(ON)
    //yes
    sum += 300.0;
    #endif

#endif


#if defined(OFF)
//no
sum += 4000.0;

#if !defined(ON)
//no
sum += 50000.0;
#endif

    //no
    sum += 0.1;
    #ifdef ON
        //no
        sum += 0.2;
    #endif

    //no
    sum += 0.01;
    #ifdef ON
        //no
        sum += 0.02;
    #else
        //no
        sum += 0.03;
    #endif

//no
    sum + 0.3;

#endif


#if !defined(OFF)
//yes
sum += 600000.0;

    #if defined(ON) && !defined(OFF)
    //yes
    sum += 80000000.0;

        #if defined(OFF) || defined(ON)
        //yes
        sum += 900000000.0;

            #if defined(ON) && defined(OFF)
                //no
                sum += 0.7;
            #elif !defined(OFF)
                //yes
                sum += 7000000.0;
            #endif

        #endif

    #endif

#endif

// sum should be 987600301.0
    gl_Position = vec4(sum);
}
