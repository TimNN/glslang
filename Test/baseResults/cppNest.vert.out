0:? Sequence
0:5  Sequence
0:5    move second child to first child (float)
0:5      'sum' (float)
0:5      0.000000
0:7  Function Definition: main( (void)
0:7    Function Parameters: 
0:12    Sequence
0:12      add second child into first child (float)
0:12        'sum' (float)
0:12        1.000000
0:21      add second child into first child (float)
0:21        'sum' (float)
0:21        300.000000
0:61      add second child into first child (float)
0:61        'sum' (float)
0:61        600000.000000
0:65      add second child into first child (float)
0:65        'sum' (float)
0:65        80000000.000000
0:69      add second child into first child (float)
0:69        'sum' (float)
0:69        900000000.000000
0:76      add second child into first child (float)
0:76        'sum' (float)
0:76        7000000.000000
0:86      move second child to first child (4-component vector of float)
0:86        'gl_Position' (gl_Position 4-component vector of float)
0:86        Construct vec4 (4-component vector of float)
0:86          'sum' (float)
0:?   Linker Objects

